src/app/clusters/ota-requestor/ota-requestor-server.cpp

```diff
@@ -258,19 +262,31 @@ bool emberAfOtaSoftwareUpdateRequestorClusterAnnounceOtaProviderCallback(
     chip::app::CommandHandler * commandObj, const chip::app::ConcreteCommandPath & commandPath,
     const chip::app::Clusters::OtaSoftwareUpdateRequestor::Commands::AnnounceOtaProvider::DecodableType & commandData)
 {
-    EmberAfStatus status;
-    chip::OTARequestorInterface * requestor = chip::GetRequestorInstance();
+    auto & metadataForNode = commandData.metadataForNode;
 
-    if (requestor != nullptr)
+    if (commandObj == nullptr)
     {
-        status = requestor->HandleAnnounceOTAProvider(commandObj, commandPath, commandData);
+        ChipLogError(SoftwareUpdate, "Invalid commandObj, cannot process AnnounceOtaProvider");
+        return false;
```

Comment:
Should we return True here? The caller will try to access "commandObj" is we return false.